packages/core/src/shared/types.ts

```diff
@@ -0,0 +1,16 @@
+import { GraphQLResolveInfo } from 'graphql';
+
+export type ID = string;
+export type Nil = undefined | null;
+export type Maybe<T> = T | Nil;
+export type Plural<T> = T | T[];
+export type Single<T> = T extends Array<infer R> ? R : T;
+export type ResolveFn<TContext> = (
```

Comment:
Do we need `ResolveFn` as new type here? maybe we can import it from graphql-tools?